client/my-sites/stats/features/modules/stats-referrers/stats-referrers.tsx

```diff
@@ -0,0 +1,91 @@
+import { StatsCard } from '@automattic/components';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { megaphone } from '@wordpress/icons';
+import clsx from 'clsx';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import {
+	isRequestingSiteStatsForQuery,
+	getSiteStatsNormalizedData,
+} from 'calypso/state/stats/lists/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import EmptyModuleCard from '../../../components/empty-module-card/empty-module-card';
+import { SUPPORT_URL } from '../../../const';
+import StatsModule from '../../../stats-module';
+import StatsModulePlaceholder from '../../../stats-module/placeholder';
+import { StatsEmptyActionSocial } from '../shared';
+
+type StatsRefeeresProps = {
+	className?: string;
+	period: string;
+	query: string;
+	moduleStrings: {
+		title: string;
+		item: string;
+		value: string;
+		empty: string;
+	};
+};
+
+const StatsRefeeres: React.FC< StatsRefeeresProps > = ( {
+	period,
+	query,
+	moduleStrings,
+	className,
+} ) => {
+	const translate = useTranslate();
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const statType = 'statsReferrers';
+
+	// TODO: sort out the state shape.
+	const requesting = useSelector( ( state: any ) =>
+		isRequestingSiteStatsForQuery( state, siteId, statType, query )
+	);
+	const data = useSelector( ( state ) =>
+		getSiteStatsNormalizedData( state, siteId, statType, query )
+	) as [ id: number, label: string ]; // TODO: get post shape and share in an external type file.
+
+	return (
+		<>
+			{ /* This will be replaced with ghost loaders, fallback to the current implementation until then. */ }
+			{ requesting && <StatsModulePlaceholder isLoading={ requesting } /> }
+			{ ( ! data || ! data?.length ) && (
```

Comment:
Minor suggestion - I think if we are using conditional chaining here, then `!data` could be omitted, couldn't it? The same applies the L76.

Reply:
I'll update this with a fix coming in a moment.